src/scenes/ProgressReports/EditForm/Steps/SubmitReportStep/ConfirmVariantDialog.tsx

```diff
@@ -0,0 +1,139 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Typography,
+} from '@mui/material';
+import { Role } from '~/api/schema/schema';
+import { PromptResponseListFragment } from '~/common/fragments';
+import { useSession } from '~/components/Session';
+import { ProgressReportEditFragment } from '../../ProgressReportEdit.graphql';
+
+interface ConfirmEmptyVariantsProps {
+  report: ProgressReportEditFragment;
+  open: boolean;
+  onContinue: () => void;
+  onCancel: () => void;
+  onUpdateConfirmStatus: () => void;
+}
+
+export const ConfirmEmptyVariants = ({
+  report,
+  open,
+  onContinue,
+  onCancel,
+  onUpdateConfirmStatus,
+}: ConfirmEmptyVariantsProps) => {
+  const { reportFile, teamNews, communityStories } = report;
+  const { session } = useSession();
+  const roles = new Set(session?.roles.value);
+  const hasReportProgressFile = !!reportFile.value;
+  const projectMgmtHeader = !hasReportProgressFile && reportFile.canEdit;
+  const hasSelectedTeamNewsPrompt = teamNews.items.length > 0;
+  const hasSelectedStoryPrompt = communityStories.items.length > 0;
+  const listItems = [];
+
+  const handleConfirmDialog = () => {
+    onContinue();
+  };
+
+  const handleCloseDialog = () => {
+    onCancel();
+  };
+
+  const getEmptyResponsesForRole = (
+    list: PromptResponseListFragment,
+    roles: Set<Role>
+  ) => {
+    if (list.items.length === 0) return [];
+    const emptyResponsesByRole = list.items[0]?.responses.filter((item) => {
+      return item.variant.responsibleRole
+        ? roles.has(item.variant.responsibleRole) &&
+            !item.response.value &&
+            item.response.canEdit
+        : false;
+    });
+    return emptyResponsesByRole ? emptyResponsesByRole : [];
+  };
+
+  const emptyTeamNewsResponsesByRole = getEmptyResponsesForRole(
+    teamNews,
+    roles
+  );
+  if (!hasSelectedTeamNewsPrompt || emptyTeamNewsResponsesByRole.length > 0) {
```

Comment:
I think these double ifs and function calls could be simplified to something like this:

```
  const listItemsData = [
    {
      key: "teamNews",
      condition: !report.teamNews.items.length || getEmptyResponsesForRole(report.teamNews, roles).length,
      label: "Team News"
    },
    {
      key: "story",
      condition: !report.communityStories.items.length || getEmptyResponsesForRole(report.communityStories, roles).length,
      label: "Story"
    }
  ];

  const listItems = listItemsData.filter(item => item.condition).map(item => (
    <li key={item.key}>{item.label}</li>
  ));
```

Otherwise looks good to me. Thanks for the opportunity to review!

Reply:
Thanks for all the help @adam-soltech , updated per your feedback.